src/anyvar/storage/snowflake.py

```diff
@@ -202,6 +202,30 @@ def __delitem__(self, name: str) -> None:
             cur.execute(f"DELETE FROM {self.table_name} WHERE vrs_id = ?;", [name])  # nosec B608
         self.conn.commit()
 
+    def wait_for_writes(self):
+        """Return true once any currently pending database modifications have been completed."""
+        if self.batch_thread is not None:
+            # short circuit if the queue is empty
+            with self.batch_thread.cond:
+                if len(self.batch_thread.pending_batch_list) == 0:
+                    return
+
+            # queue an empty batch
+            batch = []
+            self.batch_thread.queue_batch(batch)
+            # wait for the batch to be removed from the pending queue
+            while True:
+                with self.batch_thread.cond:
+                    if (
+                        len(
+                            list(filter(lambda x: x is batch, self.batch_thread.pending_batch_list))
```

Comment:
Double checking that `is` is appropriate here -- the object doesn't get copied or moved around while being held on the queue?

Reply:
Yes this should be a strict identity check.  The empty batch is essentially a sentinel object and batch objects are not modified as they move through the queue.